awesome_cart/awc.py

```diff
@@ -231,48 +231,45 @@ def get_product_by_sku(sku, detailed=0, awc_session=None, quotation=None):
 	"""Get's product in awcjs format by its sku and optionally detailed data."""
 
 	price_list = None
+
 	if is_logged_in():
 		if not quotation:
 			quotation = _get_cart_quotation()
-		price_list = quotation.get("selling_price_list")
-
-	# fetch item by its sku/item_code
-	item = frappe.get_all("Item", fields=["name", "item_name" ,"variant_of", "item_code", "disabled", "default_warehouse", "net_weight"], filters = {"item_code": sku}, ignore_permissions=1)
 
-	if not item or len(item) == 0:
-		return { "success": False, "data": None }
+		price_list = quotation.get("selling_price_list")
 
-	item = item[0]
+	try:
+		item = frappe.get_doc("Item", sku)
+	except frappe.DoesNotExistError:
+		return {"success": False, "data": None}
 
 	if item.disabled:
-		return { "success": False, "data": None }
+		return {"success": False, "data": None}
 
-	# get awc item name by its item link
-	awc_item_name = frappe.db.get_value("AWC Item", filters = {"product_name": item.name})
+	awc_item_name = frappe.db.get_value("AWC Item", {"product_name": item.name}, "name")
 
 	missing_awc_item = False
-	# lets check if we have an AWC Item for this Item
 	if not awc_item_name:
-		# if not, lets check if we have an AWC Item for its variant_of value if set
-		if item.get('variant_of'):
-			awc_item_name = frappe.db.get_value("AWC Item", filters = {"product_name": item.variant_of})
+		# let's check if item is a variant, and if we have an AWC Item for the parent item
+		if item.variant_of:
+			awc_item_name = frappe.db.get_value("AWC Item", {"product_name": item.variant_of}, "name")
+
 			if not awc_item_name:
-				missing_awc_item = "{0}(Parent), {1}(Variant)".format(item.variant_of, item.name)
+				missing_awc_item = "{0.variant_of} (Parent), {0.name} (Variant)".format(item)
 		else:
 			missing_awc_item = item.name
 
 	if missing_awc_item:
-		return { "success": False, "data": None, "error": "Missing AWC Item for {0}".format(missing_awc_item) }
+		return {"success": False, "data": None, "error": "Missing AWC Item for {0}".format(missing_awc_item)}
 
-	# finally get awc_item doctype instance
 	awc_item = frappe.get_doc("AWC Item", awc_item_name)
 	# get awc_item custom data as dictionary
 	custom_data = get_awc_item_custom_data(awc_item)
 
-	price_info = get_price(item.get("item_code"), price_list)
+	price_info = get_price(item.item_code, price_list)
 	price = price_info.get("rate")
 
-	variants = frappe.get_all("Item", fields=["name", "item_code"], filters={"variant_of": item.get("name")})
+	variants = frappe.get_all("Item", fields=["item_code"], filters={"variant_of": item.name, "disabled": 0})
```

Comment:
That name field makes me paranoid that it might be used somewhere else, just not necessarily on the listing or cart.
Can you put it back until we have more unit tests on these to quench my ocd?

Reply:
Heh, sure.